packages/plugin/src/theia.d.ts

```diff
@@ -13197,12 +13203,12 @@ export module '@theia/plugin' {
         /**
          * The shell command. Is `undefined` if created with a full command line.
          */
-        command?: string | ShellQuotedString;
+        command: string | ShellQuotedString | undefined;
```

Comment:
If I read this change correctly, it does not really change the usage of the class in any way, right? it probably would force literals to include a value for "command" if it was an interface, but not for a class.
That said, if I look at the implementation of `ShellExecution` in `types-impl.ts`, I see this:
```
set commandLine(value: string) {
        if (typeof value !== 'string') {
            throw illegalArgument('commandLine');
        }
```

So it throws an exception if I assign `undefined`. But that's not consistent with the tying. What's the reason?

Reply:
Basically, for the shell execution, we are supposed to have either commandLine set to a valid string, or the command and eventually args. 
The check in the types-impl is redundant with the one in the constructor. I would also not expect the shell execution to be changed at runtime. In case we still want to change at runtime, this may be done using an atomic change between command line and command + args, to ensure only one group is defined at a time.

Reply:
I have a `command` value, shouldn't we be checking that we don't have a `commandLine` value? Because we're not supposed to have both, right? Anyway, why aren't we respecting the interface and complaining at execution time that the user set something unusable. 
>  I would also not expect the shell execution to be changed at runtime.

But the API allows it, so people are going to do it 🤷 This check might break exensions that do 
``` 
execution.command  = undefined;
execution.commandLine = 'ls -ls`;
```
which would work perfectly fine.

Reply:
So if I understand you, we should allow `undefined` in the setCommand, setCommandLine (and setArgs), remove the checks done on the values, and allow the getters to return also `undefined`? 
Otherwise, we would have issues with the first line 
`execution.command  = undefined;`
both commandLine and command would be undefined at the same time.

Reply:
The typing says you can assign undefined to  `command`, so if the implementation is not allowing that, there needs to be a reasons for that. If we try to not allow both `command` and `commandLine`, we would have to have a different check. It seems the checks are coming from the `extHostImplementation.ts` in VS Code. But those checks would still allow for both `command` and `commandLine` to be set. So we're bug-compatible with VS Code in this respect, but it is still wrong and it still bugs me 🤷